File: api/golang/ToolGood/TextFilter/Api/Impl/TextFilterProvider.go
package Impl

 
import "encoding/json"
import "net/http"
import	"net/url"
import	. "../Datas/Requests"
import	"io/ioutil"
import	"strconv"
import	. "../Datas/Texts"


type TextFilterProvider struct{
	textFilterHost string
}

const textFilterUrl string = "/api/text-filter";
const textReplaceUrl string = "/api/text-replace";
const htmlFilterUrl string = "/api/html-filter";
const htmlReplaceUrl string = "/api/html-replace";
const jsonFilterUrl string = "/api/json-filter";
const jsonReplaceUrl string = "/api/json-replace";
const markdownFilterUrl string = "/api/markdown-filter";
const markdownReplaceUrl string = "/api/markdown-replace";

func NewTextFilterProvider(config string) *TextFilterProvider{
	provider := &TextFilterProvider{}
	provider.textFilterHost = config
	return provider;
}

func (this *TextFilterProvider)TextFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+textFilterUrl;
	return this.textFilter(request,u);
}
func (this *TextFilterProvider)HtmlFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+htmlFilterUrl;
	return this.textFilter(request,u);
}
func (this *TextFilterProvider)JsonFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+jsonFilterUrl;
	return this.textFilter(request,u);
}
func (this *TextFilterProvider)MarkdownFilter(request *TextFilterRequest) *TextFilterResult  {
	u:=this.textFilterHost+markdownFilterUrl;
	return this.textFilter(request,u);
}


func (this *TextFilterProvider)TextReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+textReplaceUrl;
	return this.textReplace(request,u);
}
func (this *TextFilterProvider)HtmlReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+htmlReplaceUrl;
	return this.textReplace(request,u);
}
func (this *TextFilterProvider)JsonReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+jsonReplaceUrl;
	return this.textReplace(request,u);
}
func (this *TextFilterProvider)MarkdownReplace(request *TextReplaceRequest) *TextReplaceResult  {
	u:=this.textFilterHost+markdownReplaceUrl;
	return this.textReplace(request,u);
}



func (this *TextFilterProvider)textReplace(request *TextReplaceRequest,u string)*TextReplaceResult   {
	resp, err := http.PostForm(u, url.Values{ "txt":{request.Txt}, "replaceChar":{request.ReplaceChar}, "reviewReplace":{strconv.FormatBool(request.ReviewReplace)}, "contactReplace":{strconv.FormatBool(request.ContactReplace)}, "skipBidi":{strconv.FormatBool(request.SkipBidi)}, "onlyPosition":{strconv.FormatBool(request.OnlyPosition)} })
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *TextReplaceResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}


func (this *TextFilterProvider)textFilter(request *TextFilterRequest,u string)*TextFilterResult   {
	resp, err := http.PostForm(u, url.Values{ "txt":{request.Txt}, "skipBidi":{strconv.FormatBool(request.SkipBidi)}, "onlyPosition":{strconv.FormatBool(request.OnlyPosition)}	}) 
    if err != nil {
		return nil
    }
	defer resp.Body.Close() 
    body,err := ioutil.ReadAll(resp.Body)
    if err != nil {
		return nil
    }
	var result *TextFilterResult
	err = json.Unmarshal([]byte(body), &result)
	if err != nil {
		return nil
	}
	return result
}

File: api/python/Impl/TextFilterProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# TextFilterProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 
 
__all__ = ['TextFilterProvider']
import json
import http.client, urllib.parse
import requests


class TextFilterProvider():


    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig
        self.__TextFilterUrl = "/api/text-filter"
        self.__TextReplaceUrl = "/api/text-replace"
        self.__HtmlFilterUrl = "/api/html-filter"
        self.__HtmlReplaceUrl = "/api/html-replace"

        self.__JsonFilterUrl = "/api/json-filter"
        self.__JsonReplaceUrl = "/api/json-replace"
        self.__MarkdownFilterUrl = "/api/markdown-filter"
        self.__MarkdownReplaceUrl = "/api/markdown-replace"
    
    def TextFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig, self.__TextFilterUrl,request)
        return json.loads(result)

    def TextReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__TextReplaceUrl,request)
        return json.loads(result)

    def HtmlFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本
        @skipBidi 是否跳过 Bidi 字符，默认 false
        @onlyPosition 只显示位置，不显示匹配文字
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__HtmlFilterUrl,request)
        return json.loads(result)

    def HtmlReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__HtmlReplaceUrl,request)
        return json.loads(result)

    def JsonFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__JsonFilterUrl,request)
        return json.loads(result)

    def JsonReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        
        result=doPost(self.__textFilterConfig,self.__JsonReplaceUrl,request)
        return json.loads(result)
 
    def MarkdownFilter(self,txt:str,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        request={'txt':txt,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__MarkdownFilterUrl,request)
        return json.loads(result)

    def MarkdownReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        request={'txt':txt,'replaceChar':replaceChar,'reviewReplace':reviewReplace,'contactReplace':contactReplace,'skipBidi':skipBidi,'onlyPosition':onlyPosition}
        result=doPost(self.__textFilterConfig,self.__MarkdownReplaceUrl,request)
        return json.loads(result)

 
 
def doPost(textFilterConfig,url,postData):
    u=textFilterConfig.GetTextFilterHost()+url
    result = requests.post(u, postData)
    return  result.text